cleverhans/future/torch/attacks/noise.py

```diff
@@ -0,0 +1,42 @@
+"""
+The Noise Attack
+"""
+
+import numpy as np
+import torch
+
+
+def noise(x, eps=0.3, order=np.inf, clip_min=None, clip_max=None):
+    """
+    A weak attack that just picks a random point in the attacker's action
+    space. When combined with an attack bundling function, this can be used to
+    implement random search.
+
+    References:
+    https://arxiv.org/abs/1802.00420 recommends random search to help identify
+        gradient masking
+
+    https://openreview.net/forum?id=H1g0piA9tQ recommends using noise as part
+        of an attack building recipe combining many different optimizers to
+        yield a strong optimizer.
+
+    Args:
+    :param x: the input tensor
+    :param eps: (optional float) maximum distortion of adversarial example 
+                compared to original input.
+    :param ord: (optional) Order of the norm.
+    :param clip_min: (optional float) Minimum input component value
+    :param clip_max: (optional float) Maximum input component value
+    """
+
+    if order != np.inf: raise NotImplementedError(ord)
+    
+    eta = torch.FloatTensor(*x.shape, device=x.device) \
+               .uniform_(-eps, eps)
+
+    adv_x = x.float() + eta
```

Comment:
What is the purpose of casting with `.float()` here? Is it required for the attack to work? (This question may be silly, I'm new to PyTorch)

Reply:
Just in case the `x` passed is of some other type like `Byte` it will throw an error.
So, its safe to do a `.float()`
```
RuntimeError: expected backend CPU and dtype Float but got backend CPU and dtype Byte
```

Reply:
@david-berthelot could you take a look?

Reply:
I wouldn't recommending casting x to float, my personal preference is to error out rather than do some magic under the hood that can be hard to debug. Ideally I'd suggest returning a tensor of the same type as was provided in input.

Reply:
I agree with you. That's a nice point, its better to throw an Exception than to change the type of the passed input internally. I will make the changes and push asap.

Reply:
Okay! Changed. You guys were right, I designed my tests in a way such that a `ByteTensor` was sent as the input, which is actually an edge case, as most of the time the inputs will be in `float` and the `eta` is in `float` already, so there is no need to cast. Sorry, my bad.